flashinfer-jit-cache/build_backend.py

```
@@ -79,23 +79,32 @@ def _create_build_metadata():
 
 def _compile_jit_cache(output_dir: Path, verbose: bool = True):
     """Compile AOT modules using flashinfer.aot functions directly."""
+    # Get the project root directory
+    project_root = Path(__file__).parent.parent
+
+    # Ensure 3rdparty submodules are populated (may be empty in CI Docker images)
+    import subprocess
+
+    subprocess.run(
+        ["git", "submodule", "update", "--init", "--recursive"],
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)